src/main/java/com/example/demo/archive/service/ArchiveService.java

```diff
@@ -66,4 +74,23 @@ public ArchiveResponse create(User author, CreateArchiveRequest request, List<Mu
 
         return ArchiveResponse.fromEntity(archive);
     }
+
+    @Transactional(readOnly = true)
+    public List<ArchiveResponse> findNearArchives(User author, double currentX, double currentY) {
+        return archiveRepository.findNearArchives(currentX, currentY)
+                .stream()
+                .map(ArchiveResponse::fromEntity)
+                .collect(Collectors.toList());
+    }
+
+    private Point generateLocation(double x, double y) {
+        String pointWKT = String.format("POINT(%s %s)", y, x);
+        Point location = null;
+        try {
+            location = (Point) new WKTReader().read(pointWKT);
+        } catch (ParseException ex) {
+            return null;
+        }
+        return location;
```

Comment:
예외 가능성 있는 parse보다 factory 사용이 더 안전해 보여요

```suggestion
        var geometryFactory = new GeometryFactory();
        return geometryFactory.createPoint(new Coordinate(x, y));
```

혹시 사용하시려면
그대로 사용하시지는 마시고 GeometryFactory Bean에 등록 해주세요

Reply:
팩토리를 쓰면 예외 사용에 안전한가요? 이유가 무엇인가요?!

Reply:
제가 너무 띄엄띄엄 적었네요..

문자열이 잘못되는경우 런타임에서 익셉션이 발생합니다.
문자열 파싱 없이 작성하면 컴파일이나 빌드에서 에러 발생하여서 에러를 빨리 알 수 있고
WKT IDE에서 지원도 없어서 IDE 도움 받기도 어려워서
유지보수 할때 휴먼 에러발생할 확율이 올라갑니다.

그것과 별개로 역직렬화 하는거보다 오브젝트 생성 바로하는게 성능면에서 좋아요